core/src/main/java/com/opensymphony/xwork2/interceptor/StaticParametersInterceptor.java

```diff
@@ -237,7 +237,8 @@ protected void addParametersToContext(ActionContext ac, Map<String, ?> newParams
             }
         } else {
             if (newParams != null) {
-                combinedParams = combinedParams.withExtraParams(newParams);
+                HttpParameters newHttpParameters = HttpParameters.create(newParams).build();
+                combinedParams = combinedParams.withParent(newHttpParameters);
             }
             if (previousParams != null) {
                 combinedParams = combinedParams.withParent(previousParams);
```

Comment:
@lukaszlenart I think for this one means for previous params, combinedParams.withParent(previousParams) needs to replace with combinedParams.withExtraParams(previousParams) otherwise extra parameters which are in newParams and not in previousParams will be lost. 
Please let me know if I'm missing anything here.

Reply:
Not sure if I got your point, yet please check the latest change. The problem is that I don't have a test case to reproduce the problem.